.github/workflows/tests.yml

```diff
@@ -0,0 +1,226 @@
+        if: startsWith(runner.os, 'macOS')
+        with:
+          path: ~/Library/Caches/pip
+          key: ${{ runner.os }}-py${{ matrix.python-version }}-pip-${{ hashFiles('requirements/*/*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-py${{ matrix.python-version }}-pip-
+      - name: Cache ccache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ccache
+          key: ${{ runner.os }}-py${{ matrix.python-version }}-ccache-${{ hashFiles('src/*') }}
+          restore-keys: |
+            ${{ runner.os }}-py${{ matrix.python-version }}-ccache-
+      - name: Cache Matplotlib
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/matplotlib
+            !~/.cache/matplotlib/tex.cache
+            !~/.cache/matplotlib/test_cache
+          key: ${{ runner.os }}-py${{ matrix.python-version }}-mpl-${{ github.ref }}-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-py${{ matrix.python-version }}-mpl-${{ github.ref }}-
+            ${{ runner.os }}-py${{ matrix.python-version }}-mpl-
+
+      - name: Install Python dependencies
+        run: |
+          # Upgrade pip and setuptools and wheel to get as clean an install as
+          # possible.
+          python -mpip install --upgrade pip setuptools wheel
+
+          # Install dependencies from PyPI.
+          python -mpip install --upgrade $PRE \
+            cycler kiwisolver numpy pillow pyparsing python-dateutil \
+            -r requirements/testing/travis_all.txt \
+            ${{ matrix.extra-requirements }}
+
+          # Install optional dependencies from PyPI.
+          # Sphinx is needed to run sphinxext tests
+          python -mpip install --upgrade sphinx
+
+          # GUI toolkits are pip-installable only for some versions of Python
+          # so don't fail if we can't install them.  Make it easier to check
+          # whether the install was successful by trying to import the toolkit
+          # (sometimes, the install appears to be successful but shared
+          # libraries cannot be loaded at runtime, so an actual import is a
+          # better check).
+          if [[ "${{ runner.os }}" != 'macOS' ]]; then
+            # PyGObject, pycairo, and cariocffi do not install on OSX 10.12;
+            # pycairo 1.20+ requires a new version of Cairo, unavailable on
+            # Ubuntu 16.04, so PyGObject must be installed without build
+            # isolation in order to pick up the lower pre-installed version.
+             python -mpip install --upgrade 'pycairo<1.20.0' 'cairocffi>=0.8' &&
+             python -mpip install --upgrade --no-build-isolation PyGObject &&
+               python -c 'import gi; gi.require_version("Gtk", "3.0"); from gi.repository import Gtk' &&
+               echo 'PyGObject is available' ||
+               echo 'PyGObject is not available'
+
+            # There are no functioning wheels available for OSX 10.12 (as of
```

Comment:
This comment is no longer relevant (as this is not running on ancient OSX).

I do not think this is a blocker, but we might want to try running this here.